src/tests/gtests.cpp

```diff
@@ -36,33 +37,55 @@
 
 using namespace Utils;
 
-// Setup global objects after command line has been parsed
-void init_global_objects() {
-    thread_pool.initialize(cfg_num_threads);
+void expect_regex(std::string s, std::string re, bool positive=true) {
+    auto m = std::regex_search(s, std::regex(re));
+    if (positive && !m) {
+        FAIL() << "Output:" << std::endl << s
+            << "Does not contain:" << std::endl
+            << re << std::endl;
+    } else if (!positive && m) {
+        FAIL() << "output:" << std::endl << s
+            << "Should not contain:" << std::endl
+            << re << std::endl;
+    }
+}
 
-    // Use deterministic random numbers for hashing
-    auto rng = std::make_unique<Random>(5489);
-    Zobrist::init_zobrist(*rng);
+class LeelaEnv: public ::testing::Environment {
+public:
+    ~LeelaEnv() {}
+    void SetUp() {
+        GTP::setup_default_parameters();
+        cfg_gtp_mode = true;
 
-    // Initialize the main thread RNG.
-    // Doing this here avoids mixing in the thread_id, which
-    // improves reproducibility across platforms.
-    Random::get_Rng().seedrandom(cfg_rng_seed);
+        // Setup global objects after command line has been parsed
+        thread_pool.initialize(cfg_num_threads);
 
-    NNCache::get_NNCache().set_size_from_playouts(cfg_max_playouts);
+        // Use deterministic random numbers for hashing
+        auto rng = std::make_unique<Random>(5489);
+        Zobrist::init_zobrist(*rng);
 
-    // Initialize network
-    // Needs a weights file
-    // Network::initialize();
-}
+        // Initialize the main thread RNG.
+        // Doing this here avoids mixing in the thread_id, which
+        // improves reproducibility across platforms.
+        Random::get_Rng().seedrandom(cfg_rng_seed);
+
+        NNCache::get_NNCache().set_size_from_playouts(cfg_max_playouts);
+
+        cfg_weightsfile = "../src/tests/0k.txt";
+        Network::initialize();
+    }
+    void TearDown() {}
+};
+
+::testing::Environment* const leela_env = ::testing::AddGlobalTestEnvironment(new LeelaEnv);
```

Comment:
Is there some way to use this without making dummy vars and relying on the initialization for side effects?

Reply:
Looks like we would have to write our own main. I'll keep that in mind but I don't see myself getting to it soon.

https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#global-set-up-and-tear-down

> However, we strongly recommend you to write your own main() and call AddGlobalTestEnvironment() there, as relying on initialization of global variables makes the code harder to read and may cause problems when you register multiple environments from different translation units and the environments have dependencies among them (remember that the compiler doesn't guarantee the order in which global variables from different translation units are initialized).